plugins/modules/managed_cluster_addon.py

```diff
@@ -0,0 +1,190 @@
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+
+module: managed_cluster_addon
+
+short_description: managed cluster addon
+
+author:
+- "Hao Liu (@TheRealHaoLiu)"
+- "Hanqiu Zhang (@hanqiuzh)"
+- "Nathan Weatherly (@nathanweatherly)"
+- "Tsu Phin Hee (@tphee)"
+
+description:
+- Use managed_cluster_addon to enable/disable an addon on a managedcluster.
+
+options:
+    hub_kubeconfig:
+        description: Path to the Hub cluster kubeconfig. Can also be specified via K8S_AUTH_KUBECONFIG environment variable.
+        type: str
+        required: True
+    wait:
+        description: Whether to wait for addon to show up.
+        type: bool
+        default: False
+        required: False
+    timeout:
+        description: Number of seconds to wait for the addon to show up.
+        type: int
+        default: 60
+        required: False
+    managed_cluster:
+        description: Name of managed cluster to enabled addon.
+        type: str
+        required: True
+    addon_name:
+        description: Name of the addon to enable/disable on a managed cluster.
+        type: str
+        choices: [  application-manager,
+                    cert-policy-controller,
+                    cluster-proxy,
+                    iam-policy-controller,
+                    managed-serviceaccount,
+                    policy-controller,
+                    search-collector
+                 ]
+        required: True
+    state:
+        description:
+        - Determines if addon should be enabled, or disabled. When set to C(present),
+          an addon will be enabled. If set to C(absent), an existing addon will be disabled.
+        type: str
+        default: present
+        choices: [ absent, present ]
+        required: False
+'''
+
+EXAMPLES = r'''
+- name: "Enabled cluster-proxy addon"
+  ocmplus.cm.managed_cluster_addon:
+    state: present
+    hub_kubeconfig: /path/to/hub/kubeconfig
+    managed_cluster: example-cluster
+    addon_name: cluster-proxy
+    wait: True
+    timeout: 120
+
+- name: "Disabled cluster-proxy addon"
+  ocmplus.cm.managed_cluster_addon:
+    state: absent
+    hub_kubeconfig: /path/to/hub/kubeconfig
+    managed_cluster: example-cluster
+    addon_name: cluster-proxy
+    wait: True
+    timeout: 60
+'''
+
+RETURN = r'''
+result:
+    description: message describing the addon enabled/disabled successfully done.
+    returned: success
+    type: str
+err:
+  description: Error message
+  returned: when there's an error
+  type: str
+  sample: null
+'''
+
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback, missing_required_lib
+from ansible_collections.ocmplus.cm.plugins.module_utils.import_utils import get_managed_cluster
+from ansible_collections.ocmplus.cm.plugins.module_utils.managed_cluster_addons import (
```

Comment:
these addon classes are not used any where?

Reply:
nvm, saw the code after. 

yeah, now I understand your concerns at scrum now. 

I also feel this style looks weird 😂  as our goal of introducing classes is to de-couple this file with individual addon classes, so explicitly import looks weird, and doesn't fit our goal perfectly.

If we can use a `*` can look better, or we can also consider a Factory method like get_addon(addon_name)? what do you think

Reply:
somehow my comments disappeared 😂 adding it again:

I understand your concern at scrum now. this pattern here looks weird, as our goal is to de-couple this file from individual addons, explicitly importing each individual of them doesn't fit our goal well. 

Using `*` is better, or I imagine we can also use some factory method? like get_addon(addon_name) which find the right class for us?

Reply:
I have tried dynamic loading the class but could not get it to work.  I do't think this is possible in ansible
```
    module = importlib.import_module(f'ansible_collections.ocmplus.cm.plugins.module_utils.managed_cluster_addons.{new_addon_name}')
    addon_class = getattr(module, new_addon_name)
    new_addon = addon_class(module, enabled, hub_client, managed_cluster_name, addon_name, wait, timeout)
```